Include/object.h

```diff
@@ -121,9 +121,13 @@ typedef struct {
 #define _PyVarObject_CAST(op) ((PyVarObject*)(op))
 
 #define Py_REFCNT(ob)           (_PyObject_CAST(ob)->ob_refcnt)
-#define Py_TYPE(ob)             (_PyObject_CAST(ob)->ob_type)
 #define Py_SIZE(ob)             (_PyVarObject_CAST(ob)->ob_size)
 
+static inline PyTypeObject* _Py_TYPE(const PyObject *ob) {
+    return ob->ob_type;
+}
+#define Py_TYPE(ob) _Py_TYPE(_PyObject_CAST_CONST(ob))
```

Comment:
While we are here, what do you think of updating Py_REFCNT and Py_SIZE as well to use _PyObject_CAST_CONST rather than _PyObject_CAST? It means that you have to add _PyVarObject_CAST_CONST (trivial macro).

Reply:
Good idea, I will open it for the sperate PR ;)